src/coreclr/jit/codegenlinear.cpp

```diff
@@ -762,11 +762,13 @@ void CodeGen::genCodeForBBlist()
                          // If a block was selected to place an alignment instruction because it ended
                          // with a jump, do not remove jumps from such blocks.
                          ,
-                         /* isRemovableJmpCandidate */ !GetEmitter()->emitIsLastInsCall() && !block->hasAlign()
+                         /* isRemovableJmpCandidate */ !GetEmitter()->emitIsLastInsCall() && !block->hasAlign() &&
+                             !compiler->fgInDifferentRegions(block, block->bbJumpDest)
 #else
 #ifdef TARGET_XARCH
                          ,
-                         /* isRemovableJmpCandidate */ !block->hasAlign()
+                         /* isRemovableJmpCandidate */ !block->hasAlign() &&
+                             !compiler->fgInDifferentRegions(block, block->bbJumpDest)
 #endif
 
```

Comment:
Since you're here, can you please make this simpler, as follows:

```
            {
                // If a block was selected to place an alignment instruction because it ended
                // with a jump, do not remove jumps from such blocks.
                // Do not remove a jump between hot and cold regions.
                bool isRemovableJmpCandidate = !block->hasAlign() &&
                             !compiler->fgInDifferentRegions(block, block->bbJumpDest);
#ifdef TARGET_AMD64
                // AMD64 requires an instruction after a call instruction for unwinding
                // inside an EH region so if the last instruction generated was a call instruction
                // do not allow this jump to be marked for possible later removal.
                isRemovableJmpCandidate = isRemovableJmpCandidate && !GetEmitter()->emitIsLastInsCall();
#endif
                inst_JMP(EJ_jmp, block->bbJumpDest, isRemovableJmpCandidate);
            }
            FALLTHROUGH;
```

Reply:
Sure thing; just re-pushed.